dds/DCPS/transport/multicast/MulticastInst.cpp

```diff
@@ -88,7 +88,8 @@ MulticastInst::load(ACE_Configuration_Heap& cf,
   GET_CONFIG_VALUE(cf, sect, ACE_TEXT("port_offset"),
                    this->port_offset_, u_short)
 
-  ACE_TString group_address_s;
+  // Explicitly initialize this string to stop gcc 11 from issuing a warning.
+  ACE_TString group_address_s(ACE_TEXT(""));
```

Comment:
What's the warning for this?  It seems odd that the compiler doesn't like it since the class has a default constructor.

Reply:
It was this one:

‘void* __builtin_memcpy(void*, const void*, long unsigned int)’ specified bound 18446744073709551615 exceeds maximum object size 9223372036854775807

The default constructor didn't do anything wrong though.  I'm not 100% certain why initializing it to an empty string makes the warning disappear.

Reply:
From what I can tell, it's warning on the operator= usage that's part of the macro expansion on the next line.  The GCC warning is that running this line from String_Base: ` this->rep_[len] = 0;` when rep_ points to the special "null string" value is invalid.  And if you only analyze the code at the level, it might make some sense...  
However, the warning is actually bogus since it's impossible to reach that line of code when rep_ points to NULL_String_.

The other error, the one with the very large numbers that points back to ACE's Configuration.cpp, seems to be a real bug in v6 which is fixed in v7.

Reply:
That seems like a reasonable explanation.  Nothing looked wrong to me, code-wise.

I just saw that ACE Configuration.cpp warning last night.  I will address that one this morning.